pkg/consumer/consumer.go

```diff
@@ -17,50 +19,36 @@ type Consumer interface {
 	Next() (library.Package, error)
 }
 
-type SQSConsumer struct {
-	client   *sqs.SQS
-	queueURL *string
+type PubSubConsumer struct {
+	subs *pubsub.Subscription
+	url  string
 }
 
-func NewSQSConsumer(sess *session.Session, queue string) (*SQSConsumer, error) {
-	client := sqs.New(sess)
-	result, err := client.GetQueueUrl(&sqs.GetQueueUrlInput{
-		QueueName: &queue,
-	})
+func NewPubSubConsumer(ctx context.Context) (*PubSubConsumer, error) {
+	queueURL := os.Getenv("OSSMALWARE_QUEUE_URL")
+	subs, err := pubsub.OpenSubscription(ctx, queueURL)
 	if err != nil {
 		return nil, err
 	}
-	consumer := &SQSConsumer{
-		client:   client,
-		queueURL: result.QueueUrl,
-	}
-	return consumer, nil
+	return &PubSubConsumer{
+		url:  queueURL,
```

Comment:
Out of curiosity - do we still need the `url` field? It seems that `Next()` now just needs the `subs` attribute.

Reply:
Nope, it's only required for the logging line. I can't find a way to get the URL back out of the `subs` object, so I saved the URL. Let me know if you'd prefer to drop that logging line or switch it to something else.